src/features/features.jl

```diff
@@ -1,20 +1,39 @@
 module FeatureDescriptor
 
-using ..ChemistryFeaturization.AbstractType: AbstractFeatureDescriptor
+using ..ChemistryFeaturization.AbstractType:
+    AbstractAtoms, AbstractFeatureDescriptor, AbstractCodec
 
 import ..ChemistryFeaturization.encodable_elements
 encodable_elements(fd::AbstractFeatureDescriptor) =
     throw(MethodError(encodable_elements, fd))
 export encodable_elements
 
+
+
 import ..ChemistryFeaturization.encode
-encode(fd::AbstractFeatureDescriptor, object_to_be_encoded) = throw(MethodError(encode, fd))
+"""
+    encode(fd::AbstractAtomFeatureDescriptor, atoms::AbstractAtoms)
+Encode `atoms` using the feature descriptor `fd`.
+"""
+encode(fd::AbstractFeatureDescriptor, atoms::AbstractAtoms) = fd(atoms)
 export encode
 
 import ..ChemistryFeaturization.decode
-decode(fd::AbstractFeatureDescriptor, encoded_feature) = throw(MethodError(decode, fd))
+"""
+    decode(fd::AbstractAtomFeatureDescriptor, encoded_feature)
+Decode `encoded_feature` using the feature descriptor `fd`.
+"""
+decode(fd::AbstractFeatureDescriptor, encoded_feature) =
+    fd.encoder_decoder(fd, encoded_feature)
```

Comment:
one thing I don't particularly like about this, is that this would require the `Codec` fields belonging to different `FeatureDescriptor` type to all be named `encoder_decoder`, which feels sticky.

Reply:
Meanwhile, the `encode` function does what it has to, without imposing any such restrictions.
I considered bringing back `EncodeOrDecode` and pass this something like `fd(encoded_feature, DECODE)`, but then that would be counterproductive, as we'd be right where we started (writing extra, repetitive code).
More of a nitpick, but that also doesn't feel like a great idea because it would either end up with us no longer having this sweet callable nature or cause non-uniform function signatures between interfaces for encode and decode (as it'd be `fd(atoms)` for encode, but `fd(encoded_feature, DECODE)`).